src/deploy/project.py

```diff
@@ -214,6 +214,37 @@ def _get_service(service):
 
         return matched_services
 
+
+    def is_release_deployed(self, release, environment_id):
+        """
+        Checks the release_id matches the `deployment:label` tag on the services
+        and tasks within those services. We check that the desiredCount of tasks 
+        matches the running count of tasks.
+        """
+        ecs_services = self.get_ecs_services(release, environment_id)
+
+        deployed_services = []
+        for _, services in sorted(ecs_services.items()):
+            for service in services:
+                tags = service["response"]["tags"]
+                desired_task_count = service["response"]["desiredCount"]
+                service_deployment_label = next(tag["value"] for tag in tags if tag["key"] == "deployment:label")
+                ecs = self._ecs(
+                    region_name=service.get('region_name'),
+                    role_arn=service.get('role_arn')
+                )
+                tasks = ecs.list_service_tasks(service)
+
+                deployed_tasks = []
+                for task in tasks:
+                    task_deployment_label = next(tag["value"] for tag in task["tags"] if tag["key"] == "deployment:label")
+                    deployed_tasks.append(task_deployment_label == service_deployment_label)
+                
+                deployed_services.append(all(deployed_tasks) and desired_task_count == len(tasks) and release["release_id"] == service_deployment_label)
```

Comment:
Ditto mismatching.

Reply:
We could - but for the clarity of what's going on, I'm not sure we gain much efficiency.

The logic of "have all the tasks in all the services been deployed" is easier than "have any of the tasks in any of the services not deployed"...